File: 1328-break-a-palindrome/1328-break-a-palindrome.go
func breakPalindrome(palindrome string) string {
    n := len(palindrome)
    
    for i := 0; i < n / 2; i++ {
        if string(palindrome[i]) != "a" {
            return palindrome[:i] + "a" + palindrome[i + 1:]
        }
    }
    
    if n > 1 {
        return strings.TrimSuffix(palindrome, "a") + "b"
    }
    
    return ""
}

File: 1328-break-a-palindrome/1328-break-a-palindrome.js
/**
 * @param {string} palindrome
 * @return {string}
 */
var breakPalindrome = function(palindrome) {
    const len = palindrome.length;
    
    for (let i = 0; i < (len >> 1); i++) {
        if (palindrome[i] !== 'a') {
            return palindrome.slice(0, i) + 'a' + palindrome.slice(i + 1);
        }
    }
    
    if (len > 1) {
        return palindrome.slice(0, -1) + 'b';
    }
    
    return '';
};